pypeit/spectrographs/mmt_bluechannel.py

```diff
@@ -0,0 +1,292 @@
+        # Required (core)
+        meta['ra'] = dict(ext=0, card='RA')
+        meta['dec'] = dict(ext=0, card='DEC')
+        meta['target'] = dict(ext=0, card='OBJECT')
+        meta['decker'] = dict(ext=0, card='APERTURE')
+        meta['dichroic'] = dict(ext=0, card='INSFILTE')
+        meta['binning'] = dict(ext=0, card=None, compound=True)
+        meta['mjd'] = dict(ext=0, card=None, compound=True)
+        meta['exptime'] = dict(ext=0, card='EXPTIME')
+        meta['airmass'] = dict(ext=0, card='AIRMASS')
+
+        # Extras for config and frametyping
+        meta['dispname'] = dict(ext=0, card='DISPERSE')
+        meta['idname'] = dict(ext=0, card='IMAGETYP')
+
+        # used for arc and continuum lamps
+        meta['lampstat01'] = dict(ext=0, card=None, compound=True)
+
+        # Ingest
+        self.meta = meta
+
+
+    def compound_meta(self, headarr, meta_key):
+        """
+
+        Args:
+            headarr: list
+            meta_key: str
+
+        Returns:
+            value
+
+        """
+        if meta_key == 'binning':
+            """
+            Binning in blue channel headers is space-separated rather than comma-separated.
+            """
+            binspec, binspatial = headarr[0]['CCDSUM'].split()
+            binning = parse.binning2string(binspec, binspatial)
+            return binning
+        elif meta_key == 'mjd':
+            """
+            Need to combine 'DATE-OBS' and 'UT' headers and then use astropy to make an mjd.
+            """
+            date = headarr[0]['DATE-OBS']
+            ut = headarr[0]['UT']
+            ttime = Time(f"{date}T{ut}", format='isot')
+            return ttime.mjd
+        elif meta_key == 'lampstat01':
+            """
+            If the comparison mirror is in, there will be a 'COMPLAMP' header entry containing the lamps
+            that are turned on. However, if the comparison mirror is out, then this header entry doesn't exist.
+            So need to test for it and set to 'Off' if it's not there.
+            """
+            if 'COMPLAMP' in headarr[0]:
+                return headarr[0]['COMPLAMP']
+            else:
+                return 'off'
+        else:
+            msgs.error(f"Not ready for compound meta, {meta_key}, for MMT Blue Channel.")
+
+
+    def default_pypeit_par(self):
+        """
+        Set default parameters for MMT/Blue Channel reductions.
+        """
+        par = pypeitpar.PypeItPar()
+        par['rdx']['spectrograph'] = 'mmt_bluechannel'
+        # Wavelengths
+        # 1D wavelength solution
+        par['calibrations']['wavelengths']['rms_threshold'] = 0.5
+        par['calibrations']['wavelengths']['sigdetect'] = 5.
+        par['calibrations']['wavelengths']['fwhm']= 5.0
+        # HeNeAr is by far most commonly used, though ThAr is used for some situations.
+        par['calibrations']['wavelengths']['lamps'] = ['ArI', 'ArII', 'HeI', 'NeI']
+        par['calibrations']['wavelengths']['method'] = 'holy-grail'
+
```

Comment:
If the tilt does not change too much for most users, I would suggest to add an arxiv solution and then use full_template. The holy-grail could fail for some data with unknown reasons.

Reply:
`holy-grail` works quite well for the 300-line grating, but doesn't for other gratings. the plan was to leave this as-is for now and implement `full_template` once i get more calibration data to support the other gratings. the 300-line and 500-line gratings are used within a fairly small range of grating angle, but the 1200-line grating has a broad blaze function and is used with central wavelengths ranging from 4200 A to 6500 A. so that will need 4 or 5 settings to cover the usable range.

Reply:
agreed